pkg/controller/mutationtemplate/mutationtemplate_controller.go

```diff
@@ -165,7 +168,7 @@ func (r *ReconcileMutationTemplate) handleCreate(instance *templatesv1alpha1.Mut
 	if !containsString(finalizerName, instance.GetFinalizers()) {
 		instance.SetFinalizers(append(instance.GetFinalizers(), finalizerName))
 		if err := r.Update(context.Background(), instance); err != nil {
-			log.Error(err, "update error", err)
+			log.Error(err, "update error")
 			return reconcile.Result{Requeue: true}, nil
```

Comment:
@maxsmythe : should we not return err here as well? I don't understand under what circumstances the Reconcile function should return a non-nil error

Reply:
returning non-nil errs implies a reconcile requeue, so I have no issue replacing `reconcile.Result{Requeue: true}` with `err`, but I'm also unsure what situations we should be returning nil and non-nil errs

Reply:
Ok, up to you and Max

Reply:
returning nil errors with requeue is when an error is expected and not noteworthy (e.g. a write conflict means we need to retry the reconcile)